### PR TITLE
Order enum variants, lower case error messages

### LINKED ISSUE #865: Cleanup Error enum
## Description

Cleanup iota.rs Error enum

## Motivation

Currently the errors are more or less just added in some way, but there is no clear ordering and naming (some end with `Error`, others not like `NotEnoughBalance`)

## Open questions (optional)

How should it look? 

## Are you planning to do it yourself in a pull request?

No.

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Order enum variants, remove unused variants, lower case error messages

## Links to any relevant issues

Fixes #865 

## Type of change

- Bug fix (a non-breaking change which fixes an issue)
- Enhancement (a non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Documentation Fix

## How the change has been tested

Running the node_api_core_get_output_metadata example returns now `Error: NotFound("https://supersecreturl.net/api/core/v2/outputs/0xb66fd384cb5755668f1890ea2e41d699db9cf32f3bc422ad3c24ffeb9c7f01d00000/metadata")`

## Change checklist

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/error.rs [hunk 1]
```diff
@@ -19,126 +19,117 @@ pub type Result<T> = std::result::Result<T, Error>;
...
+    MissingInputWithEd25519Address,
+    /// Error on API request
+    #[error("node error: {0}")]
+    NodeError(String),
+    /// The block doesn't need to be promoted or reattached
+    #[error("block ID `{0}` doesn't need to be promoted or reattached")]
+    NoNeedPromoteOrReattach(String),
+    /// The requested data was not found.
+    #[error("the requested data was not found.")]
+    NotFound,
```
Comment (Thoralf-M) [inline src/error.rs]: @thibault-martinez I don't like that the error doesn't contain any more information, what do you think about NotFound(String) where the String is the url for which we got this response?

Comment (thibault-martinez) [reply src/error.rs]: Yeah good idea 👍🏻

File: src/error.rs [hunk 2]
```diff
@@ -19,126 +19,109 @@ pub type Result<T> = std::result::Result<T, Error>;
...
-    /// Invalid MQTT topic.
-    #[error("The MQTT topic {0} is invalid")]
-    InvalidMqttTopic(String),
-    /// MQTT connection not found (all nodes MQTT's are disabled)
-    #[error("MQTT connection not found (all nodes have the MQTT plugin disabled)")]
-    MqttConnectionNotFound,
-    /// IO error
-    #[error("{0}")]
-    #[serde(serialize_with = "display_string")]
-    IoError(#[from] std::io::Error),
```
Comment (thibault-martinez) [inline src/error.rs:125]: What was the motive to remove this one?

Comment (Thoralf-M) [reply src/error.rs:125]: It wasn't used in the library, only in the examples, no?

Comment (thibault-martinez) [reply src/error.rs:125]: Hmm yeah it wasn't used by *ourselves* but we're still a library, it just makes it easier for consumers to use `?` I guess

Comment (Thoralf-M) [reply src/error.rs:125]: Yes, but we shouldn't add support for all kind of possible errors, just so it's easier to use from us, even so we don't need this at all?

Comment (thibault-martinez) [reply src/error.rs:125]: Yeah I understand the point of view. I would have said that IO at least is a good one to handle but let's forget about it 👍🏻

**Review by thibault-martinez — APPROVED** [review]

**Review by samuel-rufi — APPROVED** [review]

**Pull request merged**